packages/devtools/lib/src/memory/memory_plotly.dart

```diff
@@ -0,0 +1,192 @@
+        size: 10,
+      ),
+      name: 'GC',
+      text: [],
+      hoverinfo: 'y+name',
+    );
+    */
+
+    final Data externalTrace = Data(
+      x: [],
+      y: [],
+      text: [],
+      line: Line(
+        color: EXTERNAL_COLOR,
+      ),
+      type: 'scatter',
+      stackgroup: 'one',
+      name: 'External',
+    );
+
+    final Data usedTrace = Data(
+      x: [],
+      y: [],
+      text: [],
+      line: Line(
+        color: USED_COLOR,
+      ),
+      type: 'scatter',
+      stackgroup: 'one',
+      name: 'Used',
+    );
+
+    final Data capacityTrace = Data(
+      x: [],
+      y: [],
+      text: [],
+      line: Line(
+        color: CAPACITY_COLOR,
+        dash: 'dot',
+        width: 2,
+      ),
+      type: 'scatter',
+      mode: 'lines',
+      name: 'Capacity',
+    );
+
+    final Data rssTrace = Data(
+      x: [],
+      y: [],
+      text: [],
+      line: Line(
+        color: RSS_COLOR,
+        dash: 'dash',
+        width: 2,
+      ),
+      type: 'scatter',
+      visible: 'legendonly',
+      mode: 'lines',
+      name: 'RSS',
+    );
+
+    return [externalTrace, usedTrace, capacityTrace, rssTrace];
+  }
+
+  // Resetting to live view, it's an autoscale back to full view.
+  void _doubleClick(DataEvent data) => _memoryChart.resume();
+
+  void plotMemory() {
+    Plotly.newPlot(
+      _domName,
+      createMemoryTraces(),
+      getMemoryLayout(''),
+      Configuration(
+        responsive: true,
+        displaylogo: false,
+        displayModeBar: false,
+      ),
+    );
+
+    doubleClick(_domName, _doubleClick);
+  }
+
+  void plotMemoryDataList(
+    List<int> timestamps,
+    List<num> rsses,
+    List<num> capacities,
+    List<num> uses,
+    List<num> externals,
+  ) {
+    // TODO(terry): Eliminate this JS call (result of reified List?).
+    myExtendTraces(
+      _domName,
+      timestamps,
+      timestamps,
+      timestamps,
+      timestamps,
+      rsses,
+      capacities,
+      externals,
+      uses,
+      [
+        MEMORY_RSS_TRACE,
+        MEMORY_CAPACITY_TRACE,
+        MEMORY_EXTERNAL_TRACE,
+        MEMORY_USED_TRACE,
+      ],
+    );
+
+    if (liveUpdate) {
+      // Display 2 minutes of collected data in the chart, all data is accessible.
+      final int startTime = DateTime.fromMillisecondsSinceEpoch(timestamps[0])
+          .subtract(const Duration(minutes: 2))
+          .millisecondsSinceEpoch;
+      rangeSliderToLast(startTime, timestamps[0]);
+    }
+  }
+
+  void rangeSliderToLast(int startTime, int endTime) {
+    Plotly.update(
+      _domName,
+      [Data()],
+      Layout(
+        xaxis: AxisLayout(
+          range: [startTime, endTime],
+          rangeslider: RangeSlider(
+            autorange: true,
+          ),
+          type: 'date',
+          tickformat: '%-I:%M:%S %p',
+          hoverformat: '%-I:%M:%S.%L %p',
+        ),
+      ),
+    );
+  }
+
+  bool liveUpdate = true;
+
+  void setLiveUpdate({bool live}) {
```

Comment:
this method doesn't seem like it adds any value. 
Just set
liveUpdate = live directly.

Reply:
Done.